### ISSUE TITLE
Ngram Error

### ISSUE THREAD

**Issue by whyooooo** [NLP-Suite/NLP-Suite#1385]
When I try to exclude punctuation or stopwords for the setting of Calculation of Ngram, the terminal give me the below error message 
<img width="548" alt="Screenshot 2024-01-31 at 6 53 11 PM" src="https://github.com/NLP-Suite/NLP-Suite/assets/144286074/09f3c8e5-35cd-498e-8964-648be99edbef">
what should I do to fix

**Comment by bianshuyang** [comment]
Can you send the file that is in your folder, the options you set, and screenshot? I test with Three Little Pigs and also larger files and it has always worked.

**Comment by whyooooo** [comment]
> Can you send the file that is in your folder, the options you set, and screenshot? I test with Three Little Pigs and also larger files and it has always worked.

I used the folder of Novel_3 as input and set the setting as below 
<img width="1292" alt="Screenshot 2024-01-31 at 7 37 28 PM" src="https://github.com/NLP-Suite/NLP-Suite/assets/144286074/ad75c5b3-827a-435b-a571-e6cad4d86ddf">

**Comment by bianshuyang** [comment]
Thanks for finding this bug. The determiners is buggy for the moment. @Emory-NLP  I believe I also found out the cause of a mysterious file a month ago that you reported.

We need to change the line 62-69 in NGrams_util to:
```
def removedt(original_sentence):
    fin = open('../lib/wordLists/determiners.txt', 'r')
    determiners = list(set(fin.read().splitlines()))
    # Create a regex pattern for the determiners, case-insensitive
    # The \b ensures the match is for whole words only, avoiding partial matches within words
    dets_pattern = r'(\b(?:' + '|'.join(map(re.escape, determiners)) + r')\b)\s*'
    # Replace determiners along with the following spaces with a single space
    filtered_sentence = re.sub(dets_pattern, " ", original_sentence, flags=re.IGNORECASE)
    # Stripping leading/trailing whitespace
    final_sentence = filtered_sentence.strip()
    return final_sentence
```

**Comment by bianshuyang** [comment]
@whyooooo  In your output FOLDER please go up one time and remove anything associated with the word "cache". 
<img width="302" alt="image" src="https://github.com/NLP-Suite/NLP-Suite/assets/52821055/1d62c2f3-1844-4bed-a042-66b4774b1d32">

In the above image, the first and third must be removed as your previous calculation may affect later results.

**Comment by whyooooo** [comment]
> @whyooooo In your output FOLDER please go up one time and remove anything associated with the word "cache". <img alt="image" width="302" src="https://private-user-images.githubusercontent.com/52821055/301351655-1d62c2f3-1844-4bed-a042-66b4774b1d32.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3NDkxNDUsIm5iZiI6MTcwNjc0ODg0NSwicGF0aCI6Ii81MjgyMTA1NS8zMDEzNTE2NTUtMWQ2MmMyZjMtMTg0NC00YmVkLWEwNDItNjZiNDc3NGIxZDMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAxVDAwNTQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0MDg5MjRlNjk1MjE5YjI3OGFjY2NjZGRhOGY5OGYxMzg4YTcyNTNhZDM1NWI0Y2E3NGE4MDFmODE4MjMyZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VwL3WjUhktvIVYhdrDzTxeybYpiK5FaKfZZfkoOpfAc">
> 
> In the above image, the first and third must be removed as your previous calculation may affect later results.

Yes I did. So at this moment before the bug got fixed, we cannot select on the option of exclude ... right?

**Comment by bianshuyang** [comment]
The only non-selectable option would be determiners, however if you use a text editor and carefully replace the function as mentioned in the thread it would have worked, I assume.

**Comment by whyooooo** [comment]
> The only non-selectable option would be determiners, however if you use a text editor and carefully replace the function as mentioned in the thread it would have worked, I assume.
<img width="588" alt="Screenshot 2024-01-31 at 8 38 26 PM" src="https://github.com/NLP-Suite/NLP-Suite/assets/144286074/9ffc5195-bbc2-48b9-89ff-5dcab94c9e55">
It still have the error message if I select "lemmatize" and "exclude stop words"

**Comment by Emory-NLP** [comment]
Fixed determiners bug in release 4.6.4.

**Comment by JesNi523** [comment]
Out of curiosity, I tried to play around with the n-gram and got this message:


![image](https://github.com/NLP-Suite/NLP-Suite/assets/122462035/d45d4e57-2cc4-4ad9-9830-ef4945fb0975)

Is there potentially a file I should try to add to the directory?
(system information, windows 10, current version 4.5.8 because I couldn't update to the latest release yet).

**Comment by Emory-NLP** [comment]
The GUI image cut out the top, so I cannot see the release version. But you say you are running release 4.5.8 but the current release should be 4.6.4. The missing file problem was fixed in release 4.6.0 (see https://github.com/NLP-Suite/NLP-Suite/wiki/NLP-Suite-Release-History)
Something is fundamentally wrong with your NLP Suite installation.

**Comment by bianshuyang** [comment]
[git_ensure.bat.zip](https://github.com/NLP-Suite/NLP-Suite/files/14125389/git_ensure.bat.zip)
<img width="220" alt="image" src="https://github.com/NLP-Suite/NLP-Suite/assets/52821055/6e106a79-1589-48dc-b133-9ce4eaa7a052">

Hi, could you try unzip the attached, place it in Setup_Windows, and run only the git_ensure.bat?
Thanks!
